kpi/views/v2/data.py

```diff
@@ -376,7 +382,8 @@ def destroy(self, request, pk, *args, **kwargs):
         url_path='(enketo\/)?edit',
     )
     def enketo_edit(self, request, pk, *args, **kwargs):
-        return self._enketo_request(request, pk, action_='edit', *args, **kwargs)
+        submission_id = positive_int(pk)
```

Comment:
ideally we should not 500 if it's the client's fault, such as `http://kf.kobo.local:9000/api/v2/assets/aYgpAJH2fLhTEDjx4ApU3Y/data/haha/enketo/edit/`, which currently raises a `ValueError: invalid literal for int() with base 10: 'haha'`

Reply:
#3692